sdk/python/lib/pulumi_policy/policy.py

```diff
@@ -0,0 +1,751 @@
+                provider = self._get_provider_resource(r)
+                resource = PolicyResource(r.type, props, r.urn, r.name, opts, provider, None, [], {})
+                property_dependencies: Dict[str, List[str]] = {}
+                for k, v in r.propertyDependencies.items():
+                    property_dependencies[k] = list(v.urns)
+                intermediates.append(IntermediateStackResource(resource, r.parent, list(r.dependencies), property_dependencies))
+
+            # Create a map of URNs to resources, used to fill in the parent and dependencies
+            # with references to the actual resource objects.
+            urns_to_resources: Dict[str, PolicyResource] = {}
+            for i in intermediates:
+                urns_to_resources[i.resource.urn] = i.resource
+
+            # Go through each intermediate result and set the parent and dependencies.
+            for i in intermediates:
+                # If the resource has a parent, lookup and set it to the actual resource object.
+                if i.parent is not None and i.parent in urns_to_resources:
+                    i.resource.parent = urns_to_resources[i.parent]
+
+                # Set dependencies to actual resource objects.
+                for d in i.dependencies:
+                    if d in urns_to_resources:
+                        i.resource.dependencies.append(urns_to_resources[d])
+
+                # Set property_dependencies to actual resource objects.
+                for k in i.property_dependencies:
+                    v = i.property_dependencies[k]
+                    deps: List[PolicyResource] = []
+                    for d in v:
+                        if d in urns_to_resources:
+                            deps.append(urns_to_resources[d])
+                    i.resource.property_dependencies[k] = deps
+
+            resources: List[PolicyResource] = []
+            for i in intermediates:
+                resources.append(i.resource)
+            args = StackValidationArgs(resources)
+
+            result = policy.validate(args, report_violation)
+            if isawaitable(result):
+                loop = asyncio.new_event_loop()
+                loop.run_until_complete(result)
+                loop.close()
+
+        return proto.AnalyzeResponse(**{
+            "diagnostics": diagnostics
+        })
+
+    def GetAnalyzerInfo(self, request, context):
+        policies: List[proto.PolicyInfo] = []
+        for policy in self.__policies:
+            enforcement_level = (policy.enforcement_level if policy.enforcement_level is not None
+                                 else self.__policy_pack_enforcement_level)
+            policies.append(proto.PolicyInfo(**{
+                "name": policy.name,
+                "description": policy.description,
+                "enforcementLevel": self._map_enforcement_level(enforcement_level),
+                # TODO[pulumi/pulumi-policy#210]: Expose config schema
+            }))
```

Comment:
Since we know the names, can't we just pass as normal named params `proto.PolicyInfo(name=policy.name, ...)`?  Same for all the other calls here.

Reply:
Yep. Fixed.